cabby/rvs/generate_rvs.py

```diff
@@ -68,106 +66,28 @@ def main(argv):
-
-    if intersection > 0:
-      if intersection == 1:
-        # Filter out templates without the next intersection mention.
-        current_templates = current_templates[
-          current_templates['next intersection'] == True]
-      else:
-        # Pick templates with either blocks or intersections. X blocks = X intersection - 1.
-        # Randomly pick if the feature is a block or an intersection.
-        # If 1 - treat it as a block feature, else as an intersection.
-        if random.randint(0, 1):
-          if intersection == 2:  # one block
-            # Filter out templates without the next block mention.
-            current_templates = current_templates[
-              current_templates['next block'] == True]
-          else: # Multiple blocks.
-            # Filter out templates without mentions of the number of blocks
-            # that should be passed.
-            current_templates = current_templates[
-              current_templates['blocks'] == True]
-        else: # Multiple intersections.
-          # Filter out templates without mentions of the number of
-          # intersections that should be passed.
-          current_templates = current_templates[
-            current_templates['intersections'] == True]
-    else:
-      # Filter out templates with mentions of intersection\block.
-      current_templates = current_templates[
-        (current_templates['intersections'] == False) &
-        (current_templates['blocks'] == False) &
-        (current_templates['next intersection'] == False) &
-        (current_templates['next block'] == False)]
-
-      # From the candidates left, pick randomly one template.
-
-      choosen_template = current_templates.sample(1)['sentence'].iloc[0]
-
-      gen_instructions, entity_span = templates.add_features_to_template(
-        choosen_template, entity)
-      rvs_entity = geo_item.RVSSample.to_rvs_sample(
-        instructions=gen_instructions,
-        id=entity_idx,
-        geo_entity=entity,
-        entity_span=entity_span
-      )
-      gen_samples.append(rvs_entity)
-
-  logging.info(f"RVS generated: {len(gen_samples)}")
-
-  uniq_samples = {}
-  for gen in gen_samples:
-    uniq_samples[gen.instructions] = gen
-  logging.info(f"RVS generated: {len(gen_samples)}")
-
-  uniq_samples = {}
-  for gen in gen_samples:
-    uniq_samples[gen.instructions] = gen
-  logging.info(f"Unique RVS generated: {len(uniq_samples)}")
-
-  logging.info(
-    f"Writing {len(uniq_samples)} samples to file => " +
-    f"{FLAGS.save_instruction_path}")
-  # Save to file.
-  with open(FLAGS.save_instruction_path, 'a') as outfile:
-    for sample in uniq_samples.values():
-      json.dump(sample, outfile, default=lambda o: o.__dict__)
-      outfile.write('\n')
-      outfile.flush()
-
-  logging.info("Finished writing to file.")
+  # Split into Train, Dev and Test sets.
+  size_templates = gen_templates.shape[0]
+  train_size = round(size_templates*80/100)
```

Comment:
It would be better to have variables train_proportion and dev_proportion, and check that they are <1.0 and that the don't add up to more than 1.0. (You can then use those below for the entity splits and maintain consistency.)